app/models/team.rb

```diff
@@ -497,6 +497,9 @@ def filtered_explainers(filters = {})
     # Filter by text
     query = query.where('(title ILIKE ? OR url ILIKE ? OR description ILIKE ?)', *["%#{filters[:text]}%"]*3) if filters[:text].to_s.size > 2
 
+    # Exclude the ones already applied to a target item
+    query = query.where.not(id: ProjectMedia.find(filters[:target_id].to_i).explainer_ids) unless ProjectMedia.find_by_id(filters[:target_id].to_i).nil?
```

Comment:
I do not know if the ProjectMedia with `filters[:target_id]` exist this means that the query to find the ProjectMedia item will run twice one to check that item exists and the second one for the original query
if yes I think it's better to get the object then do our check
i.e 
pm = ProjectMedia.find_by_id(filters[:target_id].to_i)
query = query.where.not(id: pm.explainer_ids) unless pm.nil?

Reply:
Good idea, done!